internal/dag/builder.go

```diff
@@ -35,6 +35,11 @@ func (pf ProcessorFunc) Run(dag *DAG, source *KubernetesCache) {
 	}
 }
 
+type BuilderType interface {
+	// Build builds and returns a new DAG
+	Build() *DAG
+}
+
```

Comment:
I might move this into `internal/debug` where it's required, and then can rename to `DAGBuilder` or similar.

Reply:
Great suggestion. Duck interfacing blowing my mind here.